src/System.Collections/src/System/Collections/Generic/CollectionExtensions.cs

```diff
@@ -37,5 +37,39 @@ public static class CollectionExtensions
             TValue value;            
             return dictionary.TryGetValue(key, out value) ? value : defaultValue;
         }
+
+        public static bool TryAdd<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key, TValue value)
+        {
+            if (dictionary == null)
+            {
+                throw new ArgumentNullException(nameof(dictionary));
+            }
+
+            if (!dictionary.ContainsKey(key))
+            {
+                dictionary.Add(key, value);
+                return true;
+            }
+
+            return false;
+        }
+
+        public static bool Remove<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key, out TValue value)
+        {
+            if (dictionary == null)
+            {
+                throw new ArgumentNullException(nameof(dictionary));
+            }
+
+            value = default(TValue);
+
+            if (dictionary.TryGetValue(key, out var foundValue))
+            {
+                value = foundValue;
+                return true;
+            }
+
+            return false;
+        }
```

Comment:
(which also highlights that the new tests are missing verification that the value is actually removed)